server/aesdsocket.c

```diff
@@ -252,7 +272,7 @@ int main(int argc, char *argv[])
   int listen_rc=0;
   int value = 1;
   
-  fd = open(FILE_PATH, (O_RDWR|O_CREAT|O_APPEND),0744 );
+  fd = open(FILE_PATH, (O_RDWR|O_CREAT|O_APPEND),0766 );
```

Comment:
@malolasimman you can't open the file this early when using the aesdchar driver, this will prevent the driver from unloading and cause your unit test to fail.